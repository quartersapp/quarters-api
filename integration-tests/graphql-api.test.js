/* eslint-env jest */

const request = require('supertest')
const app = require('lib/app').listen()
const { User } = require('lib/db/models')
const { userFactory } = require('test-helpers/factories')
const { encode } = require('lib/auth/token-service')

let user

beforeEach(async () => {
  await User.query().truncate()
  user = await User
    .query()
    .insert(userFactory({
      id: 12345,
      email: 'test@example.com'
    }))
})

it('can be queried for the current user', async () => {
  const token = encode({ userId: user.id })

  const query = `
    {
      currentUser {
        id, email
      }
    }
  `

  const { body } = await request(app)
    .post('/graphql')
    .set('authorization', `Bearer ${token}`)
    .send({ query })
    .expect(200)

  expect(body).toMatchSnapshot()
})
