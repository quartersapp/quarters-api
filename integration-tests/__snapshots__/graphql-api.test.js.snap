// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be queried for the current user 1`] = `
Object {
  "data": Object {
    "currentUser": Object {
      "email": "test@example.com",
      "id": "12345",
    },
  },
}
`;
